packages/typescript/src/components/FunctionDeclarationJSDoc.tsx

```diff
@@ -0,0 +1,43 @@
+import { Children, For, Show } from "@alloy-js/core";
+import { JSDoc } from "./JSDoc.jsx";
+import { JSDocParagraph } from "./JSDocParagraph.jsx";
+import { JSDocParam } from "./JSDocParam.jsx";
+import { ParameterDescriptor } from "./ParameterDescriptor.js";
+
+export interface FunctionDeclarationJSDocProps {
+  doc?: string;
+  parameters?: ParameterDescriptor[] | Record<string, Children>;
+  children?: Children;
+}
+
+export function FunctionDeclarationJSDoc(props: FunctionDeclarationJSDocProps) {
+  const canHaveParamDoc = Boolean(
+    props.parameters &&
+      Array.isArray(props.parameters) &&
+      props.parameters.length > 0,
+  );
+  const parameterDescriptors =
+    Array.isArray(props.parameters) ? props.parameters : [];
+  return (
+    <Show when={Boolean(props.doc)}>
+      <JSDoc>
+        <JSDocParagraph>{props.doc}</JSDocParagraph>
+        {canHaveParamDoc ?
+          <For each={parameterDescriptors}>
```

Comment:
Are we sure this component is necessary? It feels like it might be easier to put lines 26 -> 38 into a component like `<JSDocParameters>`, and then when creating docs for a function you use the same `JSDoc` component you would for anything else, except you also place a `<JSDocParameters>` inside.

```
<JSDoc>
  <JSDocParagraph>Cool function</JSDocParagraph>
  <JSDocParameters parameters={params} />
</JSDoc>
<FunctionDeclaration ... />
```